react/RelatedProducts.js

```diff
@@ -14,14 +14,15 @@ import ShelfItem from './ShelfItem'
  */
 class RelatedProducts extends Component {
   render() {
-    const { data, productList } = this.props
-    const products =
+    const { data, productList, slug } = this.props
+    let products =
       (data &&
         !data['error'] &&
         data.product &&
         data.product.recommendations &&
         data.product.recommendations.view) ||
       []
+    products = products.filter(product => product.linkText !== slug)
```

Comment:
Shouldn´t this be handle on the serverside, like passing this on the product query?